Spotbugs fails with EQ_COMPARETO_USE_OBJECT_EQUALS issues

```
[INFO] --- spotbugs-maven-plugin:3.1.12.1:check (default-cli) @ container-jfr ---
[INFO] BugInstance size is 4
[INFO] Error size is 0
[INFO] Total bugs: 4
[ERROR] com.redhat.rhjmc.containerjfr.platform.internal.DefaultPlatformStrategy defines compareTo(Object) and uses Object.equals() [com.redhat.rhjmc.containerjfr.platform.internal.DefaultPlatformStrategy] At DefaultPlatformStrategy.java:[line 1] EQ_COMPARETO_USE_OBJECT_EQUALS
[ERROR] com.redhat.rhjmc.containerjfr.platform.internal.KubeApiPlatformStrategy defines compareTo(Object) and uses Object.equals() [com.redhat.rhjmc.containerjfr.platform.internal.KubeApiPlatformStrategy] At KubeApiPlatformStrategy.java:[line 1] EQ_COMPARETO_USE_OBJECT_EQUALS
[ERROR] com.redhat.rhjmc.containerjfr.platform.internal.KubeEnvPlatformStrategy defines compareTo(Object) and uses Object.equals() [com.redhat.rhjmc.containerjfr.platform.internal.KubeEnvPlatformStrategy] At KubeEnvPlatformStrategy.java:[line 1] EQ_COMPARETO_USE_OBJECT_EQUALS
[ERROR] com.redhat.rhjmc.containerjfr.platform.openshift.OpenShiftPlatformStrategy defines compareTo(Object) and uses Object.equals() [com.redhat.rhjmc.containerjfr.platform.openshift.OpenShiftPlatformStrategy] At OpenShiftPlatformStrategy.java:[line 1] EQ_COMPARETO_USE_OBJECT_EQUALS
[INFO] 
```

Bug descriptions found here:
https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html

This is an interesting one since the interface defines a default compareTo, while equals/hashcode would need to be done in all of it's subclasses; maybe it's ignorable.

--- a/src/main/java/com/redhat/rhjmc/containerjfr/platform/PlatformModule.java
+++ b/src/main/java/com/redhat/rhjmc/containerjfr/platform/PlatformModule.java
@@ -134,7 +134,8 @@ public abstract class PlatformModule {
                     String.format("Selected PlatformDetectionStrategy \"%s\" not found", platform));
         }
         return strategies.stream()
-                .sorted()
+                // reverse sort, higher priorities should be earlier in the stream
+                .sorted((a, b) -> Integer.compare(b.getPriority(), a.getPriority()))
                 .filter(PlatformDetectionStrategy::isAvailable)
                 .findFirst()
                 .orElseThrow();

--- a/src/main/java/com/redhat/rhjmc/containerjfr/platform/internal/PlatformDetectionStrategy.java
+++ b/src/main/java/com/redhat/rhjmc/containerjfr/platform/internal/PlatformDetectionStrategy.java
@@ -44,8 +44,7 @@ package com.redhat.rhjmc.containerjfr.platform.internal;
 import com.redhat.rhjmc.containerjfr.net.AuthManager;
 import com.redhat.rhjmc.containerjfr.platform.PlatformClient;
 
-public interface PlatformDetectionStrategy<T extends PlatformClient>
-        extends Comparable<PlatformDetectionStrategy<?>> {
+public interface PlatformDetectionStrategy<T extends PlatformClient> {
     int PRIORITY_DEFAULT = 0;
     int PRIORITY_PLATFORM = 50;
 
@@ -56,9 +55,4 @@ public interface PlatformDetectionStrategy<T extends PlatformClient>
     T getPlatformClient();
 
     AuthManager getAuthManager();
-
-    @Override
-    default int compareTo(PlatformDetectionStrategy<?> o) {
-        return Integer.compare(o.getPriority(), getPriority());
-    }
 }

